File: TP1/src/fr/ulille1/fil/odeva/Money.java
package  fr.ulille1.fil.odeva;

import java.util.HashMap;
import java.util.Map;

public class Money {
	private int value;
	private String currency;
	

	Money(int value, String currency)
	{
		this.value=value;
		this.currency=currency;
	}

	public int getValue()
	{
		return this.value;
	}

	public String getCurrency()
	{
		return this.currency;
	}

    public String toString() {
     	return this.getValue()+" ("+this.getCurrency()+")";
    }

    public boolean equals(Object o){
		if(o == null){
			throw new NullPointerException();
		}
		if(this == o){
			return true;
		}
		if(!(o instanceof Money)){
			throw new IllegalArgumentException();
		}

		Money f = (Money) o;
		if(!(this.getCurrency()).equals(f.getCurrency())){
			if(this.getCurrency().toUpperCase().equals(f.getCurrency().toUpperCase())){
				throw new IncompatibleCurrencyException(this.getCurrency(), f.getCurrency());
			} else {
				return false;
			}
		}
		if(this.getValue() == f.getValue() && this.getCurrency() != f.getCurrency()){
			return false;
		}
		if(this.getValue() != f.getValue() && this.getCurrency() == f.getCurrency()){
			return false;
		}
		if(this.getValue() != f.getValue() && this.getCurrency() != f.getCurrency()){
			return false;
		}


		return true;
	}

	public boolean equals_2(Object o){
		return getCurrency().equals(((Money) o).getCurrency());
	}

}

File: TP2_moneyJs_sinon/src/money.js
var money= (function(){
	function money(v,curr) {
		if(v < 0){
			throw new ValueNegativeException(v);
		}
		if(curr){
			if(curr.length > 3){
				throw new DeviseCaracSupException(curr);
			} else {
				curr = curr.toUpperCase();
			}
		}
		this.v=v;
		this.curr=curr;
	}

	money.prototype.getCurrency=function () {
		return this.curr;
	}
	money.prototype.getValue=function () {
		return this.v;
	}
	money.prototype.equals=function (otherM) {
		return (otherM.getValue()==this.getValue() && otherM.getCurrency().toUpperCase()==this.getCurrency().toUpperCase()) ;
	}
	money.prototype.toString=function toString() {
		return this.getValue()+" ("+this.getCurrency()+")" ;
	}


return money;
})();
